Tests/LibWeb/Text/input/HTML/Location-search-setter.html

```diff
@@ -0,0 +1,12 @@
+<!DOCTYPE html>
+<script src="../include.js"></script>
+<script>
+    test(() => {
+        try {
+            window.location.search = "?foo=bar";
```

Comment:
Setting this unconditionally will cause the document to load repeatedly. Could we conditionally set this if `window.location.search === ""`?

Reply:
Good catch! I followed your advice and it shouldn't loop indefinitely anymore. Changes are commited in https://github.com/LadybirdBrowser/ladybird/pull/1140/commits/8d7f5478c801570c0186508f93a01189209982c2.
Hope it's okay.
Thank you!